Update Norway holidays: update subdivision list as per 2024 administrative reform

Exerqtor opened the pull request:
<!--
  Thanks for contributing to holidays!
-->

## Proposed change

<!--
  Describe the big picture of your changes.
  Don't forget to link your PR to an existing issue if any.
-->

Viken county was disolved in 2024 with Østfold, Akershus & Buskerud being reinstated.

## Type of change

<!--
  Type of change you want to introduce. Please, check one (1) box only!
  If your PR requires multiple boxes to be checked, most likely it needs to
  be split into multiple PRs.
-->

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new `holidays` functionality in general)

## Checklist

<!--
  Put an `x` in the boxes that apply. You can change them after PR is created.
-->

- [x] I've read and followed the [contributing guidelines](https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.md).
- [x] I've run `make check` locally; all checks and tests passed.

<!--
  Thanks again for your contribution!
-->

Exerqtor committed:
Update norway.py

Viken was disolved in 2024 with Østfold, Akershus & Buskerud being reinstated.

Signed-off-by: Exerqtor <41635644+Exerqtor@users.noreply.github.com>

PPsyrius reviewed (commented):
You'll need to do the following as well:
- Update Norway's section in the readme file
- Add Viken to the `_deprecated_subdivisions` list in the `norway py` file; see https://github.com/vacanza/holidays/blob/dev/holidays/countries/united_states.py#L237 or https://github.com/vacanza/holidays/blob/dev/holidays/countries/france.py#L102 for example 
- Add `test_subdiv_deprecation` and `setUp(self)` override to `test_norway.py` file; see https://github.com/vacanza/holidays/blob/dev/tests/countries/test_france.py#L71 for example

As far as I know, the official ISO-3166 entry for NO hasn’t been updated with the new counties list yet: https://www.iso.org/obp/ui/#iso:code:3166:NO - do you have references for these new county ISO codes?

Exerqtor commented:
Turns out it's been even more changes to the counties farther North as well, so i have to take a better look at it soon.

Regarding the ISO i have no idea, but it do look like haven't been updated for some reason. Even though the counties have been officialy changed by the Norwegian goverment (linked site in Norwegian): https://www.regjeringen.no/no/tema/kommuner-og-regioner/kommunestruktur/nye-kommune-og-fylkesnummer-fra-1.-januar-2024/id2924701/

Exerqtor committed:
Update norway.py

"Vestfold og Telemark" and "Troms og Finnmark" also got split up into four respective counties in 2024.

Signed-off-by: Exerqtor <41635644+Exerqtor@users.noreply.github.com>

Exerqtor committed:
Update norway.py

Added `_deprecated_subdivisions` list.

Signed-off-by: Exerqtor <41635644+Exerqtor@users.noreply.github.com>

Exerqtor committed:
Update test_norway.py

Added `setUp(self)` & `test_subdiv_deprecation`.

Signed-off-by: Exerqtor <41635644+Exerqtor@users.noreply.github.com>

Exerqtor committed:
Update README.md

Updated Norwegian counties.

Signed-off-by: Exerqtor <41635644+Exerqtor@users.noreply.github.com>

PPsyrius commented on holidays/countries/norway.py:
```diff
@@ -56,19 +60,30 @@ class Norway(HolidayBase, ChristianHolidays, InternationalHolidays):
+        "38",
+        "54",
+    )
+    
```
From `make pre-commit`:
```suggestion

```

PPsyrius commented on tests/countries/test_norway.py:
```diff
@@ -21,6 +21,13 @@ class TestNorway(CommonCountryTests, SundayHolidays, TestCase):
     def setUpClass(cls):
         super().setUpClass(Norway)
```
You need to `import warnings` at the very top as well, see French test case example here:
```python
import warnings
from unittest import TestCase

from holidays.countries.france import France, FR, FRA
from tests.common import CommonCountryTests
```

PPsyrius committed:
Remove trailing whitespaces

Signed-off-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>

PPsyrius committed:
Add missing `warnings` import

Signed-off-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>

PPsyrius committed:
Add missing newline

Signed-off-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>

PPsyrius committed:
Fix Norway's subdivision list in readme file

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>

KJhellico commented:
The code is LGTM, but I am concerned about the absence of these changes in ISO.

Exerqtor commented:
Haven't had time to do the changes during the work week, but @PPsyrius beat me to it and fixed it. Thx!

Regarding the ISO it isn't much i can do obviously and i haven't got an answer for why it haven't been updated.

Regardless if that, "ISO 3166-2:NO" is in this point of time "outdated" / wrong when compared to the state/government enforced changes which where put in effect 1 January 2024.

KJhellico reviewed (dismissed):
LGTM.

Exerqtor committed:
Merge branch 'dev' into patch-1

arkid15r committed:
Update code
